catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryFunction.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.util.impl;
+
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+/**
+ * Functional interface used to abstract away methods that take in a {@link QueryRequest} and
+ * return a {@link QueryResponse}. Used by classes such as {@link ResultIterable} to support
+ * interfaces other than {@link ddf.catalog.CatalogFramework}.
+ */
+@FunctionalInterface
+public interface QueryFunction {
+    QueryResponse query(QueryRequest queryRequest)
```

Comment:
✏️ I know in our earlier pre-review of this code I had suggested this interface should be updated to return `QueryResponse` instead of the more generic `SourceResponse`; however, if someone were to want to use the new iterables with a query against `CatalogFacade`, this would no longer work. I think it should go back to `SourceResponse`.